### ISSUE TITLE
Repeated running of `Test` hits recursion depth trap

### ISSUE THREAD

**Issue by ChrisJefferson** [gap-system/gap#1176]
It varies for different tests, but many tests if repeated will cause a 'recursion depth trap'. For example:

```
gap> for i in [1..10000] do Test("tst/testinstall/trans.tst"); Print(i,"\n"); od;
....
trans.tst
msecs: 746
29
########> Diff in tst/testinstall/trans.tst:45
# Input is:
TransformationListListNC([1, 2, 3], [4, 5, 6]);
# Expected output:
Transformation( [ 4, 5, 6, 4, 5, 6 ] )
# But found:
Error, recursion depth trap (5000)
########
trans.tst
msecs: 739
30
```

I think this is being caused by longjmp/setjmp not storing/restoring the stack depth (if we fix this, we could consider storing/restoring the value of `~` at the same time, and wonder if there is anything else).

**Comment by fingolfin** [comment]
It would be great if somebody could make a PR to fix this -- and don't get distracted by "other things that would need to be fixed"... if you are concerned about those, file a new issue about that -- but there is no point in delaying a fix for an issue just because there is a hypothetical chance of other things being broken...

Of course it is a very reasonable concern, and likely to be true, but digging into that and fixing it properly sounds like a much larger task than just fixing the issue at hand...

**Comment by fingolfin** [comment]
This was resolved in PR #1753 by @ChrisJefferson (we really should make it a habit to mention issues affected by PRs in the PRs, ideally as "resolves #NNN`, so that we don't have resolved issues lingering around).
